hera_cal/io.py

```diff
@@ -776,6 +776,34 @@ def get_file_lst_range(filepaths, filetype='uvh5', add_int_buffer=False):
     return file_starts, file_stops, int_times
 
 
+def partial_time_io(hd, times, **kwargs):
+    '''Perform partial io with a time-select on a HERAData object, even if it is intialized
+    using multiple files, some of which do not contain any of the specified times.
+    
+    Arguments:
+        hd: HERAData object intialized with (usually multiple) uvh5 files
+        times: list of times in JD to load
+        kwargs: other partial i/o kwargs (see io.HERAData.read)
+
+    Returns:
+        data: DataContainer mapping baseline keys to complex visibility waterfalls
+        flags: DataContainer mapping baseline keys to boolean flag waterfalls
+        nsamples: DataContainer mapping baseline keys to interger Nsamples waterfalls
+        '''
+    combined_hd = None
+    for f in hd.filepaths:
+        hd_here = HERAData(f)
+        times_here = [t for t in times if t in hd_here.times]
```

Comment:
can we have it catch the case of files that don't have partial io and thus `hd_here.times == None` and have it just load all times?

Reply:
It's not clear that that's what we want it to do. I'll add an assertion instead.